Mage.Sets/src/mage/cards/t/TheFourthDoctor.java

```diff
@@ -0,0 +1,220 @@
+            + "<i>(Artifacts, legendaries, and Sagas are historic.)</i>";
+    }
+
+    private TheFourthDoctorPlayFromTopEffect(final TheFourthDoctorPlayFromTopEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public TheFourthDoctorPlayFromTopEffect copy() {
+        return new TheFourthDoctorPlayFromTopEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        throw new IllegalArgumentException("Wrong code usage: can't call applies method on empty affectedAbility");
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability affectedAbility, Ability source, Game game, UUID playerId) {
+
+        Card cardToCheck = game.getCard(objectId);
+        if (cardToCheck == null) {
+            return false;
+        }
+
+        // to let the player cast a spell
+        if (affectedAbility instanceof SpellAbility) {
+            SpellAbility spellAbility = (SpellAbility) affectedAbility;
+            if (spellAbility.getManaCosts().isEmpty()
+                    || !spellAbility.spellCanBeActivatedNow(playerId, game).contains(MageIdentifier.Default)) {
+                return false;
+            }
+            cardToCheck = spellAbility.getCharacteristics(game);
+        }
+
+        Player controller = game.getPlayer(source.getControllerId());
+        TheFourthDoctorWatcher watcher = game.getState().getWatcher(TheFourthDoctorWatcher.class);
+        Permanent sourcePermanent = source.getSourcePermanentIfItStillExists(game);
+        Player cardOwner = game.getPlayer(cardToCheck.getOwnerId());
+
+        if (controller == null 
+                || cardOwner == null
+                || sourcePermanent == null 
+                || watcher == null) {
+            return false;
+        }
+
+        // only applies to the controller of the ability.
+        if (!playerId.equals(source.getControllerId())) {
+            return false;
+        }
+        // has the ability already been used this turn by the player?
+        if (watcher.isAbilityUsed(controller.getId(), new MageObjectReference(sourcePermanent, game))) {
+            return false;
+        }
+        Card topCard = controller.getLibrary().getFromTop(game);
+        // is the card attempted to be played the top card of the library?
+        if (topCard == null || !topCard.getId().equals(cardToCheck.getMainCard().getId())) {
+            return false;
+        }
+        
+        // check for historic land or spell and create food token.
+        if (cardToCheck.isHistoric(game)) {
+            ReflexiveTriggeredAbility ability = new ReflexiveTriggeredAbility(
+                    new CreateTokenEffect(new FoodToken()), false);
+            game.fireReflexiveTriggeredAbility(ability, source);
```

Comment:
wtf, you don't even check playable conditional -- that's very bad... cont effect's applies must not generate any data cause it calls multiple times from multiple places.

Reply:
Nope, play from not own hand check can be called at any time from any place with any params. It works in playable check and in real activating. Same for mana, protection and many other cont and rules changing effects. It's ok to be called in playable, but it must process it correctly (the better way -- works same in playable and in real -- that's why cont effects, triggers and watchers must not change game state in most use cases).

Reply:
> If that is mandatory, it would make sense to add check in fireReflexifeTriggedAbility (and other similar methods) that would identify bad usage at runtime. So when tested, the check will tell the dev that it should not be done that way.

BTW there are already exists game state consistency checking. But game still generate dirty data (e.g. duplicated triggers). Disabled by default. See `TriggeredAbilities.enableIntegrityCheckXXX` for multiple use cases and checks  (related to #8426)